config/metrics.js

```diff
@@ -1,140 +1,153 @@
-'use strict';
+"use strict";
 
 /* eslint-disable no-magic-numbers */
 /* eslint-disable sort-keys */
 
-const symbols = require('../src/symbols');
+const symbols = require("../src/symbols");
 const getBPRules = (item) => item.defaultBranchRef?.branchProtectionRule;
+const calcRestrictedSourcePercentage = (item = {}) =>
+  item?.requiredStatusChecks?.length > 0
+    ? (item.requiredStatusChecks.filter((check) => check.app !== null)?.length /
```

Comment:
the numerator here can be `undefined` which would result in `NaN` for the percentage. What happens when *none* of the checks have a source?

Reply:
We are only checking the status checks if they are present. When none of the checks have source it will give 100 percent acc. to formula.

Reply:
right but i mean, what about when there *are* requiredStatusChecks, but all of them have a null `.app`?

Reply:
It will give 0.

Reply:
ahhh right. then the `?.` isn't needed?
```suggestion
    ? (item.requiredStatusChecks.filter((check) => check.app !== null).length /
```